lib/router/router.js

```diff
@@ -127,16 +135,21 @@ export default class Router extends EventEmitter {
       data, props, error
     } = await this.getRouteInfo(route, pathname, query)
 
-    if (error) {
+    if (error && error.cancelled) {
       this.emit('routeChangeError', error, as)
-      throw error
+      return false
     }
```

Comment:
With this, I think it make no sense to emit `routeChangeError`.
And we could simply avoid sending all abort events.